packages/@ngtools/webpack/src/loader.ts

```diff
@@ -201,6 +202,15 @@ export function ngcLoader(source: string) {
       })
       .then(() => {
         if (plugin.typeCheck) {
+          // Check all diagnostics from this and reverse dependencies also.
+          if (!plugin.firstRun) {
+            this._module.reasons
+              .filter(reason => reason.module && reason.module instanceof NormalModule)
+              .forEach(reason => plugin.diagnose(reason.module.resource));
```

Comment:
Will this also get reverse dependencies of reverse dependencies? e.g. `index.ts` re-exporting classes.